tests/lib/lte_lc/src/main.c

```diff
@@ -0,0 +1,242 @@
+	zassert_true(fabs(15.36 - cfg.ptw) < 0.1, "Wrong PTW value");
+
+	err = parse_edrx(at_response_ltem_2, &cfg);
+	zassert_equal(0, err, "parse_edrx failed, error: %d", err);
+	zassert_true(fabs(20.48 - cfg.edrx) < 0.1, "Wrong eDRX value");
+	zassert_true(fabs(19.2 - cfg.ptw) < 0.1, "Wrong PTW value");
+	zassert_false(fabs(81.92 - cfg.edrx) < 0.1, "Wrong eDRX value");
+	zassert_false(fabs(15.36 - cfg.ptw) < 0.1, "Wrong PTW value");
+
+	err = parse_edrx(at_response_nbiot, &cfg);
+	zassert_equal(0, err, "parse_edrx failed, error: %d", err);
+	zassert_true(fabs(2621.44 - cfg.edrx) < 0.1, "Wrong eDRX value");
+	zassert_true(fabs(20.48 - cfg.ptw) < 0.1, "Wrong PTW value");
+
+	err = parse_edrx(at_response_nbiot_2, &cfg);
+	zassert_equal(0, err, "parse_edrx failed, error: %d", err);
+	zassert_true(fabs(2621.44 - cfg.edrx) < 0.1, "Wrong eDRX value");
+	zassert_true(fabs(15.36 - cfg.ptw) < 0.1, "Wrong PTW value");
+}
+
+void test_parse_cereg(void)
+{
+	int err;
+	enum lte_lc_nw_reg_status status;
+	enum lte_lc_lte_mode mode;
+	struct lte_lc_cell cell;
+	struct lte_lc_psm_cfg psm_cfg;
+	char *at_response_0 = "+CEREG: 5,0,,,9,0,0,,";
+	char *at_response_1 = "+CEREG: 5,1,\"0A0B\",\"01020304\",9,0,0,\"00100110\",\"01011111\"";
+	char *at_response_2 = "+CEREG: 5,2,\"0A0B\",\"01020304\",9,0,0,\"11100000\",\"11100000\"";
+	char *at_response_3 = "+CEREG: 5,3,\"0A0B\",\"01020304\",9,0,0,,";
+	char *at_response_4 = "+CEREG: 5,4,\"0A0B\",\"FFFFFFFF\",9,0,0,,";
+	char *at_response_5 = "+CEREG: 5,5,\"0A0B\",\"01020304\",9,0,0,\"11100000\",\"11100000\"";
+	char *at_response_8 = "+CEREG: 5,8,\"0A0B\",\"01020304\",9,0,0,,";
+	char *at_response_90 = "+CEREG: 5,90,,\"FFFFFFFF\",,,,,";
+	char *at_response_wrong = "+CEREG: 5,10,,\"FFFFFFFF\",,,,,";
+	char *at_notif_0 = "+CEREG: 0,\"FFFF\",\"FFFFFFFF\",9,0,0,,";
+	char *at_notif_1 = "+CEREG: 1,\"0A0B\",\"01020304\",9,0,0,\"00100110\",\"01011111\"";
+	char *at_notif_2 = "+CEREG: 2,\"0A0B\",\"01020304\",9,0,0,,";
+	char *at_notif_3 = "+CEREG: 3,\"0A0B\",\"01020304\",9,0,0,,";
+	char *at_notif_4 = "+CEREG: 4,\"FFFF\",\"FFFFFFFF\",9,0,0,,";
+	char *at_notif_5 = "+CEREG: 5,\"0A0B\",\"01020304\",9,0,0,\"11100000\",\"00011111\"";
+	char *at_notif_8 = "+CEREG: 8,\"0A0B\",\"01020304\",9,0,0,,";
+	char *at_notif_90 = "+CEREG: 90,,\"FFFFFFFF\",,,,,";
+	char *at_notif_wrong = "+CEREG: 10,,\"FFFFFFFF\",,,,,";
+
+	/* For CEREG reads, we only check the network status, as that's the only
+	 * functionality that is exposed.
+	 */
+	err = parse_cereg(at_response_0, false, &status, NULL, NULL, NULL);
+	zassert_equal(0, err, "parse_cereg failed, error: %d", err);
+	zassert_equal(status, LTE_LC_NW_REG_NOT_REGISTERED,
+		      "Wrong network registation status");
+
+	err = parse_cereg(at_response_1, false, &status, &cell, &mode, &psm_cfg);
+	zassert_equal(0, err, "parse_cereg failed, error: %d", err);
+	zassert_equal(status, LTE_LC_NW_REG_REGISTERED_HOME,
+		      "Wrong network registation status (%d)", status);
```

Comment:
Linux kernel coding style advice against printing numbers in parentheses. But, this does not seem to been strictly enforced throughout the SDK.